src/components/common/formFields/Widgets.tsx

```diff
@@ -0,0 +1,366 @@
+import React, { useEffect, useRef, useState } from 'react'
+import ReactSelect, { GroupBase, StylesConfig } from 'react-select'
+import Tippy from '@tippyjs/react'
+import { ReactComponent as Error } from '../../../assets/icons/ic-warning.svg'
+import { ReactComponent as Info } from '../../../assets/icons/info-filled.svg'
+import { DropdownIndicator, getCommonSelectStyle, Option } from '../../v2/common/ReactSelect.utils'
+import { Checkbox, CheckboxProps } from './Checkbox'
+import { ConditionalWrap } from '../helpers/Helpers'
+import { OptionType } from '../../app/types'
+import './Widgets.scss'
+
+interface WidgetPropsType {
+    title: string
+    description?: string
+    rootClassName?: string
+    onChange?: (e) => void
+    onBlur?: (e) => void
+    minLength?: number
+    maxLength?: number
+    isRequired?: boolean
+    placeholder?: string
+    errorMessage?: string
+}
+
+const renderValidationErrorLabel = (message: string): JSX.Element => {
+    return (
+        <div className="error-label flex left align-start fs-11 fw-4 mt-6">
+            <span className="error-label-icon">
+                <Error className="icon-dim-16" />
+            </span>
+            <span className="ml-4 cr-5">{message}</span>
+        </div>
+    )
+}
+
+const renderDescription = (description: string) => {
+    return (
+        <div className="info-label flex left align-center fs-11 fw-4 mt-4">
+            <span className="info-label-icon">
+                <Info className="icon-dim-16" />
+            </span>
+            <span className="ml-4 cn-7">{description}</span>
+        </div>
+    )
+}
+
+interface StyledFieldPropsType {
```

Comment:
Can we extract all the interfaces in separate type file?

Reply:
Done